### PR TITLE
Minor E2E fixes

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What this PR does / why we need it**: Small fixes I found while working on #646:
- we were provisioning 2 kind clusters, "capz" with `make kind-create` and then "kind-capz" with the e2e framework. Removes the `make kind-create` command from the script and renames the e2e management cluster to `capz-e2e` for clarity
- Removes the duplicate docker build in the test-e2e make target, only build and push once
- Moves common patches in test templates to common patches directory

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by devigned** [comment]
hmm... perhaps, when running e2e local, we should not try to push the images since the test framework is loading them.

What do you think about having a `make test-e2e-local` without push and `make test-e2e` which requires `docker-push` like https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/cba51f21d701a4c783ccde2368728a3b99ea8dff/Makefile#L163?

**Comment by CecileRobertMichon** [comment]
@devigned how is it different from before? we're still setting `REGISTRY="localhost:5000/ci-e2e"` as the REGISTRY for LOCAL_ONLY

Before the kind cluster was created before the push but then it wasn't used by the actual tests

**Comment by CecileRobertMichon** [comment]
Oh I see what you mean now
```
docker push localhost:5000/ci-e2e/cluster-api-azure-controller-amd64:20200915011819
The push refers to repository [localhost:5000/ci-e2e/cluster-api-azure-controller-amd64]
Get http://localhost:5000/v2/: dial tcp [::1]:5000: connect: connection refused
```

File: Makefile [hunk 1]
```diff
@@ -160,7 +160,17 @@ test-integration: ## Run integration tests
 	go test -v -tags=integration ./test/integration/...
 
 .PHONY: test-e2e
-test-e2e: docker-build docker-push $(ENVSUBST) $(KUBECTL) $(GINKGO) ## Run e2e tests
+test-e2e: $(ENVSUBST) $(KUBECTL) $(GINKGO) ## Run e2e tests
+	PULL_POLICY=IfNotPresent $(MAKE) docker-build docker-push
```
Comment (CecileRobertMichon) [inline Makefile:164]: we were running docker build twice, this might also help shave off a few minutes on e2e time

**Comment by nader-ziada** [comment]
/lgtm

@devigned do you have any comments?

**Pull request merged**